src/marqo/instance_mappings.py

```diff
@@ -0,0 +1,33 @@
+from abc import ABC, abstractmethod
+
+from typing import Dict, Optional
+
+
+class InstanceMappings(ABC):
+    """
+    Abstract class for mapping index names to URLs.
+
+    Implementations of this class must ensure at least asymptotically constant computational complexity for all methods.
+    An inefficient implementation of this class can cause the Marqo client to be slow.
+    Index names must be unique across all instances.
```

Comment:
Let's try to be more helpful:

"Index names must be unique across all instances." -> "The namespace of index names is with respect to a Client instance. Index names must be unique across all Marqo URLs referenced in the Mappings. If you want to support the same index name on different Marqo URLs, please use separate Client objects. "